File: backend/src/main/java/csc13001/plantpos/domain/models/Staff.java
package csc13001.plantpos.domain.models;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name = "staffs")
public class Staff {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "staff_id")
    private Long staffId;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "user_id")
    private User user;

    @Column(name = "name", length = 256)
    private String name;

    @Column(name = "phone", length = 20)
    private String phone;

    @Column(name = "email", length = 256)
    private String email;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    public void updateSales(Double sales) {
    }
}

File: csc13001-plant-pos.Core/Models/Staff.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace csc13001_plant_pos.Core.Models;

[Table("staff")]
public class Staff
{
    [Key]
    [Column("staff_id")]
    public int StaffId
    {
        get; set;
    }

    [Column("user_id")]
    [ForeignKey("user_id")]
    public int UserId
    {
        get; set;
    }

    [Column("name")]
    [StringLength(256)]
    public string Name
    {
        get; set;
    }

    [Column("phone")]
    [StringLength(20)]
    public string Phone
    {
        get; set;
    }

    [Column("email")]
    [StringLength(256)]
    public string Email
    {
        get; set;
    }

    [Column("address")]
    public string Address
    {
        get; set;
    }
}
